### PR TITLE
Add validations and unit tests

### PR DESCRIPTION
**Pull request by Tamunokuro**
- [x] Add the following validations:
  - [x] For the User model:
    - Name must not be blank.
    - PostsCounter must be an integer greater than or equal to zero.
  - [x] For the Post model:
    - Title must not be blank.
    - Title must not exceed 250 characters.
    - CommentsCounter must be an integer greater than or equal to zero.
    - LikesCounter must be an integer greater than or equal to zero.
- [x] Add unit specs for all of your models' methods and validations.

### CODE REVIEW

File: spec/models/user/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  it 'postscounter must be an integer' do
+    subject.postscounter = 'string'
+    expect(subject).to_not be_valid
+  end
+
+  it 'postscounter must be greater than or equal to 0' do
+    subject.postscounter = -1
+    expect(subject).to_not be_valid
+  end
+end
```
Comment (andersonlebon) [inline spec/models/user/user_spec.rb]: - [ ] Kindly note that all custom methods on models should be tested also, Kindly make sure the `most_recent_posts` is tested here

File: spec/models/post/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  it 'likes_counter must be an integer' do
+    subject.likes_counter = 'string'
+    expect(subject).to_not be_valid
+  end
+
+  it 'likes_counter must be greater than or equal to 0' do
+    subject.likes_counter = -1
+    expect(subject).to_not be_valid
+  end
+end
```
Comment (andersonlebon) [inline spec/models/post/post_spec.rb]: - [ ] Kindly note that all custom methods in models should be tested also, Kindly make sure the `recent_comments` and `posts_counter` is tested here.

File: spec/models/comment/comment_spec.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  it 'is not valid without a user_id' do
+    subject.author_id = nil
+    expect(subject).to_not be_valid
+  end
+
+  it 'is not valid without a post_id' do
+    subject.post_id = nil
+    expect(subject).to_not be_valid
+  end
+end
```
Comment (andersonlebon) [inline spec/models/comment/comment_spec.rb]: - [ ] Kindly note that all custom methods in models should be tested also, Kindly make sure the `comments_counter` and `decrement_comments_counter` are tested here.

File: spec/models/post/post_spec.rb (hunk 2 above)
Comment (andersonlebon) [inline spec/models/post/post_spec.rb]: - [ ] Kindly note that all custom methods in models should be tested also, Kindly make sure the `recent_comments` and `posts_counter` are tested here.

File: spec/models/like/like_spec.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  it 'is not valid without a user_id' do
+    subject.author_id = nil
+    expect(subject).to_not be_valid
+  end
+
+  it 'is not valid without a post_id' do
+    subject.post_id = nil
+    expect(subject).to_not be_valid
+  end
+end
```
Comment (andersonlebon) [inline spec/models/like/like_spec.rb]: - [ ] Kindly note that all custom methods in models should be tested also, Kindly make sure the `likes_counter` is tested here.

**Review by andersonlebon — CHANGES_REQUESTED** [review]
## Hi @Tamunokuro 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there

## Required Changes ♻️
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question `@andersonlebon` so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AshakaE — APPROVED** [review]
Hi @Tamunokuro , 

### Highlights ✨ 

- Thanks for making the requested changes 👍 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Proceed to the next level 👏 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**